.github/workflows/ci.yaml

```diff
@@ -7,49 +7,44 @@ on:
 jobs:
   ci:
     name: Format, lint, and test
-    runs-on: ubuntu-18.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-18.04
+            binary_target: x86_64-unknown-linux-musl
+            needs_musl: true
+          - os: ubuntu-18.04
+            binary_target: x86_64-unknown-linux-gnu
+          - os: macos-10.15
+            binary_target: x86_64-apple-darwin
+          - os: windows-2019
+            binary_target: x86_64-pc-windows-msvc
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
       - uses: davidB/rust-cargo-make@v1
 
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          profile: minimal
-          override: true
-
-      - name: Set up Makefile.toml
+      - name: Install musl tools
+        if: matrix.needs_musl
         run: |
-          cat >Makefile.toml <<EOF
-          [env]
-          CARGO_MAKE_EXTEND_WORKSPACE_MAKEFILE = true
-          CARGO_MAKE_RUN_CLIPPY = true
-          CARGO_MAKE_CLIPPY_ARGS = "--all-features -- -D warnings"
-          CARGO_MAKE_RUN_CHECK_FORMAT = true
-          RUSTFLAGS="-D warnings"
+          sudo apt-get install musl-tools
+          sudo ln -s /usr/bin/musl-gcc /usr/bin/musl-g++
```

Comment:
This extra step is needed to compiler C++ files with musl. Seems odd that `musl-g++` isn't installed by default...

Reply:
It is a bit odd, but at least there's precedent: https://github.com/emk/rust-musl-builder/blob/master/Dockerfile#L74